Add 5.15.0 changelog (#17174)
hashicorp/terraform-provider-google: CHANGELOG.md

* compute: added `stack_type` attribute for `google_compute_interconnect_attachment` resource. ([#17139](https://github.com/hashicorp/terraform-provider-google/pull/17139))
* compute: updated the `google_compute_security_policy` resource's `json_parsing` field to accept the value `STANDARD_WITH_GRAPHQL` ([#17097](https://github.com/hashicorp/terraform-provider-google/pull/17097))
* memcache: added `reserved_ip_range_id` field to `google_memcache_instance` resource ([#17101](https://github.com/hashicorp/terraform-provider-google/pull/17101))
* netapp: added `deletion_policy` field to `google_netapp_volume` resource ([#17111](https://github.com/hashicorp/terraform-provider-google/pull/17111))

BUG FIXES:
* alloydb: fixed an issue where `database_flags` in secondary `google_alloydb_instance` resources would cause a diff, as they are copied from the primary ([#17128](https://github.com/hashicorp/terraform-provider-google/pull/17128))
* filestore: made `google_filestore_instance.source_backup` field configurable ([#17099](https://github.com/hashicorp/terraform-provider-google/pull/17099))

> * vmwareengine: fixed a bug to prevent recreation of existing PCs when upgrading provider version from <5.10.0 ([#17135](https://github.com/hashicorp/terraform-provider-google/pull/17135)

Reviewer: ```suggestion
* vmwareengine: fixed a bug to prevent recreation of existing [`google_vmwareengine_private_cloud`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/vmwareengine_private_cloud) resources when upgrading provider version from <5.10.0 ([#17135](https://github.com/hashicorp/terraform-provider-google/pull/17135)
```

* **New Resource:** `google_compute_region_network_endpoint` ([#17137](https://github.com/hashicorp/terraform-provider-google/pull/17137))
* **New Resource:** `google_discovery_engine_chat_engine` ([#17145](https://github.com/hashicorp/terraform-provider-google/pull/17145))
* **New Resource:** `google_discovery_engine_search_engine` ([#17146](https://github.com/hashicorp/terraform-provider-google/pull/17146))
* **New Resource:** `google_netapp_volume_snapshot` ([#17138](https://github.com/hashicorp/terraform-provider-google/pull/17138))

IMPROVEMENTS:
* compute: added `INTERNET_IP_PORT` and `INTERNET_FQDN_PORT` options for the `google_compute_region_network_endpoint_group` resource. ([#17137](https://github.com/hashicorp/terraform-provider-google/pull/17137))
* compute: added `creation_timestamp` to `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager`. ([#17110](https://github.com/hashicorp/terraform-provider-google/pull/17110))
* compute: added `disk_id` attribute to `google_compute_disk` resource ([#17112](https://github.com/hashicorp/terraform-provider-google/pull/17112))

> * compute: added `enabled` field to `compute_backend_service.iap` resource (revert) ([#17118](https://github.com/hashicorp/terraform-provider-google/pull/17118))

Reviewer: this was a revert

```suggestion
```

* compute: added `stack_type` attribute for `google_compute_interconnect_attachment` resource. ([#17139](https://github.com/hashicorp/terraform-provider-google/pull/17139))

> * compute: updated the `google_compute_security_policy` resource's `json_parsing` field to accept the value STANDARD_WITH_GRAPHQL ([#17097](https://github.com/hashicorp/terraform-provider-google/pull/17097))

Reviewer: ```suggestion
* compute: updated the `google_compute_security_policy` resource's `json_parsing` field to accept the value `STANDARD_WITH_GRAPHQL` ([#17097](https://github.com/hashicorp/terraform-provider-google/pull/17097))
```

* memcache: added `reserved_ip_range_id` field to `google_memcache_instance` resource ([#17101](https://github.com/hashicorp/terraform-provider-google/pull/17101))
* netapp: added `deletion_policy` field to `google_netapp_volume` resource ([#17111](https://github.com/hashicorp/terraform-provider-google/pull/17111))

BUG FIXES:
* alloydb: fixed an issue where `database_flags` in secondary `google_alloydb_instance` resources would cause a diff, as they are copied from the primary ([#17128](https://github.com/hashicorp/terraform-provider-google/pull/17128))

> * filestore: made source_backup field configurable ([#17099](https://github.com/hashicorp/terraform-provider-google/pull/17099))

Reviewer: ```suggestion
* filestore: made `google_filestore_instance.source_backup` field configurable ([#17099](https://github.com/hashicorp/terraform-provider-google/pull/17099))
```

> * vmwareengine: fixed a bug to prevent recreation of existing PCs when upgrading provider version from <5.10.0 ([#17135](https://github.com/hashicorp/terraform-provider-google/pull/17135)

Reviewer: This removed a default value from the resource; should it be under BREAKING CHANGES even though it was a bugfix?

Reviewer: should probably also reference the resource name instead of `PCs`

Author: No, this shouldn't have changed the API resource that is created.
